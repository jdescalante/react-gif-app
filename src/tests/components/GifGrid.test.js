import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Puebas en el componente GifGrid', () => {
    const category = 'Marvel';
    
    test('debe de mostrar el componente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot()
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: 'abc123',
            url: 'https://localhost/imagen.jpg',
            title: 'Titulo de una pruena'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow( <GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
