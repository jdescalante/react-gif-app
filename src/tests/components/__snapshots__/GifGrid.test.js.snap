// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en el componente GifGrid debe de mostrar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Marvel
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Puebas en el componente GifGrid debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Marvel
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc123"
      key="abc123"
      title="Titulo de una pruena"
      url="https://localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;
